napari/layers/labels/labels.py

```diff
@@ -190,6 +194,12 @@ def __init__(
         self._background_label = 0
         self._selected_label = 1
         self._selected_color = self.get_color(self._selected_label)
+        if color_dict is None:
+            self.color_dict = {}
+            self.enable_custom_color_dict = False
+        else:
+            self.color_dict = color_dict
```

Comment:
I think `color_dict` should be a property with a setter, and when it is set it should trigger the necessary updates (like colormap and refresh) so that data then uses the new colormap.

Reply:
I think the two `color_dict` updates that you have inside `color_dict_to_colormap` right now should be inside the color_dict setter and should only occur if those keys aren't present, i.e. we shouldn't overwrite values someone gives us.

```python
    if self._background_label not in color_dict:
          color_dict[background_label] = 'transparent'
    if None not in color_dict:
          color_dict[None] = empty_color
```